tests/Agent/IntegrationTests/IntegrationTests/AzureFunction/AzureFunctionTests.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTests.RemoteServiceFixtures;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.IntegrationTests.AzureFunction
+{
+    public abstract class AzureFunctionTestsBase<TFixture> : NewRelicIntegrationTest<TFixture>
+        where TFixture : AzureFunctionApplicationFixture
+    {
+        private readonly TFixture _fixture;
+
+        protected AzureFunctionTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+
+            _fixture.AddActions(
+                setupConfiguration: () =>
+                {
+                    new NewRelicConfigModifier(fixture.DestinationNewRelicConfigFilePath)
+                        .ForceTransactionTraces()
+                        .ConfigureFasterTransactionTracesHarvestCycle(20)
+                        .ConfigureFasterMetricsHarvestCycle(15)
+                        .ConfigureFasterSpanEventsHarvestCycle(15)
+                        .SetLogLevel("finest");
+                },
+                exerciseApplication: () =>
+                {
+                    _fixture.Get("api/function1");
+                    _fixture.AgentLog.WaitForLogLines(AgentLogBase.TransactionSampleLogLineRegex, TimeSpan.FromMinutes(2));
+                }
+            );
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var expectedTransactionEventAttributes = new List<string>
+            {
+                "faas.coldStart",
+                "faas.invocation_id",
+                "faas.name",
+                "faas.trigger",
+                "cloud.resource_id"
+            };
+
+            var expectedMetrics = new List<Assertions.ExpectedMetric>()
+            {
+                new() {metricName = "DotNet/Function1", callCount = 1},
+                new() {metricName = "DotNet/Function1", metricScope = "WebTransaction/AzureFunction/Function1", callCount = 1},
+                new() {metricName = "WebTransaction/AzureFunction/Function1", callCount = 1},
+                new() {metricName = "WebTransactionTotalTime/AzureFunction/Function1", callCount = 1},
+                new() {metricName = "Apdex/AzureFunction/Function1", callCount = 1},
+            };
+
+
+            var metrics = _fixture.AgentLog.GetMetrics().ToList();
+            Assertions.MetricsExist(expectedMetrics, metrics);
+
+            //var transactionSample = _fixture.AgentLog.TryGetTransactionSample("WebTransaction/AzureFunction/Function1");
+            //Assert.NotNull(transactionSample);
```

Comment:
Is this a todo for a future test?

Reply:
Kind of. Until I (or someone) figures out the interaction with aspnetcore instrumentation, the sampled transaction won't be the function transaction. But the integration test implementation isn't complete at this point either - I just wanted to get something working.